src/Microsoft.AspNetCore.Razor.Tools/GenerateCommand.cs

```diff
@@ -0,0 +1,204 @@
+                File.WriteAllText(outputFilePath, result.CSharpDocument.GeneratedCode);
+            }
+
+            return success ? 0 : -1;
+        }
+
+        private IReadOnlyList<TagHelperDescriptor> GetTagHelpers(string tagHelperManifest)
+        {
+            if (!File.Exists(tagHelperManifest))
+            {
+                return Array.Empty<TagHelperDescriptor>();
+            }
+
+            using (var stream = File.OpenRead(tagHelperManifest))
+            {
+                var reader = new JsonTextReader(new StreamReader(stream));
+
+                var serializer = new JsonSerializer();
+                serializer.Converters.Add(new RazorDiagnosticJsonConverter());
+                serializer.Converters.Add(new TagHelperDescriptorJsonConverter());
+
+                return serializer.Deserialize<IReadOnlyList<TagHelperDescriptor>>(reader);
+            }
+        }
+
+        private List<SourceItem> GetRazorFiles(string projectDirectory, string[] sources)
+        {
+            var trimLength = projectDirectory.EndsWith("/") ? projectDirectory.Length - 1 : projectDirectory.Length;
+
+            var items = new List<SourceItem>(sources.Length);
+            for (var i = 0; i < sources.Length; i++)
+            {
+                var fullPath = Path.Combine(projectDirectory, sources[i]);
+                if (fullPath.StartsWith(projectDirectory, StringComparison.OrdinalIgnoreCase))
+                {
+                    var viewEnginePath = fullPath.Substring(trimLength).Replace('\\', '/');
+                    items.Add(new SourceItem(fullPath, viewEnginePath));
+                }
+            }
+
+            return items;
+        }
+
+        private OutputItem[] GenerateCode(RazorTemplateEngine templateEngine, IReadOnlyList<SourceItem> sources)
+        {
+            var outputs = new OutputItem[sources.Count];
+            Parallel.For(0, outputs.Length, new ParallelOptions() { MaxDegreeOfParallelism = 4 }, i =>
+            {
+                var source = sources[i];
+
+                var csharpDocument = templateEngine.GenerateCode(source.ViewEnginePath);
+                outputs[i] = new OutputItem(source, csharpDocument);
+            });
+
+            return outputs;
+        }
+
+        private struct OutputItem
+        {
+            public OutputItem(
+                SourceItem viewFileInfo,
+                RazorCSharpDocument cSharpDocument)
+            {
+                ViewFileInfo = viewFileInfo;
+                CSharpDocument = cSharpDocument;
+            }
+
+            public SourceItem ViewFileInfo { get; }
+
+            public RazorCSharpDocument CSharpDocument { get; }
+        }
+
+        private struct SourceItem
+        {
+            public SourceItem(string fullPath, string viewEnginePath)
+            {
+                FullPath = fullPath;
+                ViewEnginePath = viewEnginePath;
+            }
+
+            public string FullPath { get; }
```

Comment:
You never use this. Can represent this entire object as a `string viewEnginePath` 😉

Reply:
Will address as part of #1953